### PR TITLE
Feature/4497 technical support functions

### PR DESCRIPTION
**Pull request by whiskey**
## Description
This PR provides a base for user uploaded (error) logs. It also enables sharing of these logs to friends and family via the share menu. **Note** you have to enable this particular logging first. Unlike development builds, this log does not work without permission.

## Link to Jira
* [EXPOSUREAPP-4497](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4497)
* [EXPOSUREAPP-5782](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5782)

## Screenshots
* App-Informations:
Bildschirmfoto 2021-04-28 um 15 34 17
Bildschirmfoto 2021-04-28 um 15 34 24
* ViewController
Bildschirmfoto 2021-04-28 um 15 34 29
Bildschirmfoto 2021-04-28 um 15 34 32
Bildschirmfoto 2021-04-28 um 15 35 06
Bildschirmfoto 2021-04-28 um 15 35 11
* Start logging was tapped:
Bildschirmfoto 2021-04-28 um 15 35 17
Bildschirmfoto 2021-04-28 um 15 35 21
* Upload was tapped -> Consent Screen:
Bildschirmfoto 2021-04-28 um 15 35 27
Bildschirmfoto 2021-04-28 um 15 35 30
* Save locally was tapped -> Export Screen:
Bildschirmfoto 2021-04-28 um 15 35 47
Bildschirmfoto 2021-04-28 um 15 35 50
* History Screen:
Bildschirmfoto 2021-04-28 um 15 37 28
Bildschirmfoto 2021-04-28 um 15 37 34

### CODE REVIEW

**Comment by marcussc** [comment]
Moving the pb.swift files, is not a good Idea IMHO, now we have the files in two different places.

File: src/xcode/ENA/TestPlans/AllTests.xctestplan [hunk 1]
```diff
@@ -76,6 +76,7 @@
     {
       "parallelizable" : true,
       "skippedTests" : [
+        "ELSSubmissionTests\/testELSAuthentication()",
```
Comment (marcussc) [inline src/xcode/ENA/TestPlans/AllTests.xctestplan:79]: ?

**Review by marcussc — APPROVED** [review]
Would have been nicer to split this monster up into smaller PRs, like we usually do with our Story PRs. 
So my Approval is more like "wow yeah lots of changes, seems to look right"

**Comment by whiskey** [comment]
> Moving the pb.swift files, is not a good Idea IMHO, now we have the files in two different places.

Tbh, the whole protobuff handling sucks. The moving was an attempt to get some structure into them and maybe this caused this even bigger mess. I'll check it again.

**Review by naveeddotio — APPROVED** [review]

**Comment by naveeddotio** [comment]
> Would have been nicer to split this monster up into smaller PRs, like we usually do with our Story PRs.
> So my Approval is more like "wow yeah lots of changes, seems to look right"

Yes we did it like that, so most of the code is already reviewed. It’s that story branch plus _some_ new stuff from Carsten :)

File: src/xcode/ENA/TestPlans/AllTests.xctestplan (hunk 1 above)
Comment (whiskey) [reply src/xcode/ENA/TestPlans/AllTests.xctestplan:79]: This test does no longer exists. Must be an artifact 🤷‍♂️

Comment (whiskey) [reply src/xcode/ENA/TestPlans/AllTests.xctestplan]: It's out now.

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift [hunk 2]
```diff
@@ -950,5 +1110,33 @@ private extension URLRequest {
...
+
+		// init form
+		try body.append("\r\n--\(boundary)\r\n")
+		// prevent potential file collisions on backend
+		try body.append("Content-Disposition:form-data; name=\"file\"; filename=\"ErrorLog-\(UUID().uuidString).zip\"\r\n")
+		try body.append("Content-Type:application/zip\r\n")
+		try body.append("Content-Length: \(logData.count)\r\n")
+		try body.append("\r\n")
+		body.append(logData)
+		try body.append("--\(boundary)--\r\n")
```
Comment (mlenkeit) [inline src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift:1136]:
```suggestion
		try body.append("\r\n")
		try body.append("--\(boundary)--\r\n")
```

Comment (mlenkeit) [reply src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift:1136]: Fails against the mock server without that change

Comment (pascal-brause) [reply src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift]: Indeed, fixes the 500 error.

**Pull request merged**